packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -71,29 +95,39 @@ export class B3Propagator implements HttpTextPropagator {
   extract(context: Context, carrier: unknown, getter: GetterFunction): Context {
     const traceIdHeader = getter(carrier, X_B3_TRACE_ID);
```

Comment:
There are so many variables now that it is difficult to figure what is happening here. Can you create some helper functions which will help you to extract only the needed variables and then refactor this function into something like (cleaner and more readable)
```
const traceId = this._extractTraceId(carrier, getter);
const spanId = this._extractSpanId(carrier, getter);
const traceFlags = this._extractTraceFlags(carrier, getter);
context = this._updateContext(context, carrier, getter);
if (isValidTraceId(traceId) && isValidSpanId(spanId)) {
  return setExtractedSpanContext(context, {
    traceId,
    spanId,
    isRemote: true,
    traceFlags,
  });
}
return context;
```

Reply:
Sure, I'm not sure if having `_extractTraceId(carrier, getter)`, `_extractSpanId(carrier, getter)`, and the like actually makes the code cleaner because there'd be a lot of functions doing a lot of the same behavior. I'll just generalize it to one function and have a special case for traceFlags.

Reply:
done, thanks